pdm/_types.py

```diff
@@ -1,9 +1,10 @@
+import sys
 from typing import Dict, List, NamedTuple, Tuple, Union
 
-try:
+if sys.version_info < (3, 8):
```

Comment:
this pattern has two advantages

- mypy understands `if sys.version_info...` imports
- it becomes far easier to see when this import logic should be removed (ie when python < 3.8 support is dropped)